docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
+ 7.  ResourceCatalogEntries should not be restricted to specific claim realization mechanism.
+
+To achieve the ACL related goals, we can create namespaced resources that are selected by label at a cluster level.
+
+
+# API Objects
+
+## ResourceCatalog
+ResourceCatalog is a cluster-scoped resource.  It has a cluster-admin controlled spec that is used to drive a controller.
+The spec includes a label selector that is used in a list/watch to filter all ResourceCatalogEntries in the cluster.
+The controller will find matching entries and copy a reference into the ResourceCatalog.Status field.  The controller
```

Comment:
> Can you give an example?

Let's say the status on this object because a label selected LIST of ResourceCatalogEntries to avoid blowing out etcd.  What is resourceVersion of the ResourceCatalog object?  That object exists in etcd (objectmeta and spec), but it is a conglomeration of objects too. Is the resourceVersion the max(catalog.resourceversion, max(allEntries.resourcVersion)? if I do that, updates will be almost impossible and any cache would be busted since you could have different objects with the same resourceversion.  Also, those resourceversions are for separate resources, so ordering is not guaranteed (different etcd cases).  

Would I include both and tokenize, catalog.resourceversion + "-" + max(allentries.resourceVersion)?  That would require custom REST storage to handle updates.  What would I do for a watch?  Is 10-15 greater or less than 9-17?

It's non-trivial to resolve.  You could try tokenizing, breaking everyone and declaring an absolute ordering.